browse/services/object_store/fileobj.py

```diff
@@ -37,7 +59,7 @@ def exists(self) -> bool:
         pass
 
     @abstractmethod
-    def open(self,  *args, **kwargs) -> IO:  # type: ignore
+    def open(self, mode:str) -> BinaryMinimalFile:
```

Comment:
"pass" does not return a BinaryMinimalFile - what type do you really want here?

Reply:
`IO` is a crazy pile of signatures. It didn't seem usable with mypy. So I made a minimal interface via a `Protocol` to make things easier.